app/models/campaign.rb

```diff
@@ -30,20 +37,57 @@ class Campaign < ApplicationRecord
   validates :inbox_id, presence: true
   validates :title, presence: true
   validates :message, presence: true
+  validate :validate_campaign_inbox
+  validate :prevent_completed_campaign_from_update, on: :update
   belongs_to :account
   belongs_to :inbox
   belongs_to :sender, class_name: 'User', optional: true
 
+  enum campaign_type: { ongoing: 0, one_off: 1 }
```

Comment:
@pranavrajs @nithindavid @muhsin-k  Thoughts on better names for the campaign types? 

ongoing = ['dynamic', 'live']
one_off= ['onetime']

Reply:
On second thoughts and after going through https://english.stackexchange.com/questions/47013/word-phrase-to-mean-something-that-just-happens-once 

the current word choices don't seem so bad.